### PR TITLE
util: remove libevent <= 2.0.18 back-compat code

### PR DESCRIPTION
**Pull request by fanquake**
Now that we require libevent >=2.0.21, remove backwards compatibility code for older versions.

### CODE REVIEW

**Comment by practicalswift** [comment]
Concept ACK

**Review by hebasto — APPROVED** [review]
ACK 58857560fcd299ae4aa03250298acfe63b26c305, I've verified that there are no other places in the code with unneeded libevent back-compat code.

**Comment by fanquake** [comment]
Guix builds:
```bash
670bc438033084003d015f23ed58cc92d065c0c962d9cc753104d32b250e0a1d  guix-build-58857560fcd2/output/aarch64-linux-gnu/SHA256SUMS.part
f27f69c7e77538aa01713f202d4db89f413c55204cd1fc99ed31f2081d564116  guix-build-58857560fcd2/output/aarch64-linux-gnu/bitcoin-58857560fcd2-aarch64-linux-gnu-debug.tar.gz
390c3166f4fa47b15a4dcbfa5205d69458fb8a1c1862de956f850d047782d816  guix-build-58857560fcd2/output/aarch64-linux-gnu/bitcoin-58857560fcd2-aarch64-linux-gnu.tar.gz
d6c12f3a1b22a4cea299f660c59a50c51061594d19ef0e087583dbece913cd69  guix-build-58857560fcd2/output/arm-linux-gnueabihf/SHA256SUMS.part
c04dc4071b2295f74b2381850a361550e481a99851bc435a184f52cd85cd5ed0  guix-build-58857560fcd2/output/arm-linux-gnueabihf/bitcoin-58857560fcd2-arm-linux-gnueabihf-debug.tar.gz
56ebdfb44035d6dd4d90d5445386deefd2d27d65d1bd2c75b69317f6e951b18a  guix-build-58857560fcd2/output/arm-linux-gnueabihf/bitcoin-58857560fcd2-arm-linux-gnueabihf.tar.gz
facf87168d70a7ace1d1b368ce2bb58e1390d1ea78e6850a2bef5e31d55b3c7d  guix-build-58857560fcd2/output/dist-archive/bitcoin-58857560fcd2.tar.gz
895f1d8006935500fd2fdcc9bb994270059b881cad37ebd0654bfa5800fed113  guix-build-58857560fcd2/output/powerpc64-linux-gnu/SHA256SUMS.part
b1a422795ac9f8f16b3e52c23861f617e49ef74fdc2c6ebaef6af906f04fff81  guix-build-58857560fcd2/output/powerpc64-linux-gnu/bitcoin-58857560fcd2-powerpc64-linux-gnu-debug.tar.gz
8b42e3a20dba091011bee1cb72e9f494f88ea781c94124e1f70fd2c22797cbb8  guix-build-58857560fcd2/output/powerpc64-linux-gnu/bitcoin-58857560fcd2-powerpc64-linux-gnu.tar.gz
6d67c99cc4beccc8ea48e579b8b4f75dc47a91597e335074d806535c8def958b  guix-build-58857560fcd2/output/powerpc64le-linux-gnu/SHA256SUMS.part
f366d52438137962322a4883c1883666e97169488f0b9ddc21ce067e3944396c  guix-build-58857560fcd2/output/powerpc64le-linux-gnu/bitcoin-58857560fcd2-powerpc64le-linux-gnu-debug.tar.gz
7cf04fa623e217bcccbb42220487f298009d460dd5cb471a8f3bae22ef3e9802  guix-build-58857560fcd2/output/powerpc64le-linux-gnu/bitcoin-58857560fcd2-powerpc64le-linux-gnu.tar.gz
3a9a8990aeb0c56a7b584d1823886c73990645770659a63c45e4b85d1c3f8d65  guix-build-58857560fcd2/output/riscv64-linux-gnu/SHA256SUMS.part
5f8f663779b3f7196da9f1c9b40be7b2a8e2b33fdb6baf673520c0b07b23ce26  guix-build-58857560fcd2/output/riscv64-linux-gnu/bitcoin-58857560fcd2-riscv64-linux-gnu-debug.tar.gz
3f43cf17a71e05506822fdbacfd8b7f3af9c739fd131842014dd80c92062d566  guix-build-58857560fcd2/output/riscv64-linux-gnu/bitcoin-58857560fcd2-riscv64-linux-gnu.tar.gz
3f89ecc812a1cd0d6009a119957b9bc5831f37249153b5fafe3268e4dd3bb59c  guix-build-58857560fcd2/output/x86_64-apple-darwin18/SHA256SUMS.part
a828f7a623d38d549992af3035254ddde5a26e62638fe23a9bd76d2867683ce4  guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx-unsigned.dmg
d24624964b6b68bfb03c88f49a22170f6fb0c16f4974949cde830be9b780b7f9  guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx-unsigned.tar.gz
7e4f0ca9b199a33acdabdfb7d6d6ae6524f1d5f3bb924951d55b792e3ad19dad  guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx64.tar.gz
bb466bd2a42c2914bf70b3b27335123eccad39b569d334c6f69ff08459c8ee6b  guix-build-58857560fcd2/output/x86_64-linux-gnu/SHA256SUMS.part
430bc8eb7506c1041db17b2393927f82be44921c00f4a698477ab3a7cdd31313  guix-build-58857560fcd2/output/x86_64-linux-gnu/bitcoin-58857560fcd2-x86_64-linux-gnu-debug.tar.gz
ab04cde29a2febcad6f50f288f6bbf026fc1a2c3309cd8f6276548d79c1d422e  guix-build-58857560fcd2/output/x86_64-linux-gnu/bitcoin-58857560fcd2-x86_64-linux-gnu.tar.gz
014a940eadbcba286cac1f2df5e6d6ea1d21a3623d98d8b1300ea44ac0e9efc5  guix-build-58857560fcd2/output/x86_64-w64-mingw32/SHA256SUMS.part
99a38df35244d34ae8d247953e6f5903713d6098e66a5a41d509074dc216b926  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win-unsigned.tar.gz
a981dcde57666283d6b3bb9b0bbe2b1e54cb2b25b7301f3ecd3c5819acfd9823  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64-debug.zip
862d38ba5b4120fb55f9b4a664379c1e2ccc2940bc54e46bb8884e9f77245e81  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64-setup-unsigned.exe
539046ae22daf063c96fefe88359b9d3a85ea5f37fd27c56abd5bcb98490e290  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64.zip
```

File: src/bitcoin-cli.cpp [hunk 1]
```diff
@@ -25,6 +25,7 @@
 #include <functional>
 #include <memory>
 #include <optional>
+#include <stdexcept>
```
Comment (kristapsk) [inline src/bitcoin-cli.cpp:28]: Why this additional include? It isn't related to the removal of libevent compatibility code. If because of `std::runtime_error` use below, should be a separate commit.

**Comment by curtcurt87** [comment]
670bc438033084003d015f23ed58cc92d065c0c962d9cc753104d32b250e0a1d guix-build-58857560fcd2/output/aarch64-linux-gnu/SHA256SUMS.part f27f69c7e77538aa01713f202d4db89f413c55204cd1fc99ed31f2081d564116 guix-build-58857560fcd2/output/aarch64-linux-gnu/bitcoin-58857560fcd2-aarch64-linux-gnu-debug.tar.gz 390c3166f4fa47b15a4dcbfa5205d69458fb8a1c1862de956f850d047782d816 guix-build-58857560fcd2/output/aarch64-linux-gnu/bitcoin-58857560fcd2-aarch64-linux-gnu.tar.gz d6c12f3a1b22a4cea299f660c59a50c51061594d19ef0e087583dbece913cd69 guix-build-58857560fcd2/output/arm-linux-gnueabihf/SHA256SUMS.part c04dc4071b2295f74b2381850a361550e481a99851bc435a184f52cd85cd5ed0 guix-build-58857560fcd2/output/arm-linux-gnueabihf/bitcoin-58857560fcd2-arm-linux-gnueabihf-debug.tar.gz 56ebdfb44035d6dd4d90d5445386deefd2d27d65d1bd2c75b69317f6e951b18a guix-build-58857560fcd2/output/arm-linux-gnueabihf/bitcoin-58857560fcd2-arm-linux-gnueabihf.tar.gz facf87168d70a7ace1d1b368ce2bb58e1390d1ea78e6850a2bef5e31d55b3c7d guix-build-58857560fcd2/output/dist-archive/bitcoin-58857560fcd2.tar.gz 895f1d8006935500fd2fdcc9bb994270059b881cad37ebd0654bfa5800fed113 guix-build-58857560fcd2/output/powerpc64-linux-gnu/SHA256SUMS.part b1a422795ac9f8f16b3e52c23861f617e49ef74fdc2c6ebaef6af906f04fff81 guix-build-58857560fcd2/output/powerpc64-linux-gnu/bitcoin-58857560fcd2-powerpc64-linux-gnu-debug.tar.gz 8b42e3a20dba091011bee1cb72e9f494f88ea781c94124e1f70fd2c22797cbb8 guix-build-58857560fcd2/output/powerpc64-linux-gnu/bitcoin-58857560fcd2-powerpc64-linux-gnu.tar.gz 6d67c99cc4beccc8ea48e579b8b4f75dc47a91597e335074d806535c8def958b guix-build-58857560fcd2/output/powerpc64le-linux-gnu/SHA256SUMS.part f366d52438137962322a4883c1883666e97169488f0b9ddc21ce067e3944396c guix-build-58857560fcd2/output/powerpc64le-linux-gnu/bitcoin-58857560fcd2-powerpc64le-linux-gnu-debug.tar.gz 7cf04fa623e217bcccbb42220487f298009d460dd5cb471a8f3bae22ef3e9802 guix-build-58857560fcd2/output/powerpc64le-linux-gnu/bitcoin-58857560fcd2-powerpc64le-linux-gnu.tar.gz 3a9a8990aeb0c56a7b584d1823886c73990645770659a63c45e4b85d1c3f8d65 guix-build-58857560fcd2/output/riscv64-linux-gnu/SHA256SUMS.part 5f8f663779b3f7196da9f1c9b40be7b2a8e2b33fdb6baf673520c0b07b23ce26 guix-build-58857560fcd2/output/riscv64-linux-gnu/bitcoin-58857560fcd2-riscv64-linux-gnu-debug.tar.gz 3f43cf17a71e05506822fdbacfd8b7f3af9c739fd131842014dd80c92062d566 guix-build-58857560fcd2/output/riscv64-linux-gnu/bitcoin-58857560fcd2-riscv64-linux-gnu.tar.gz 3f89ecc812a1cd0d6009a119957b9bc5831f37249153b5fafe3268e4dd3bb59c guix-build-58857560fcd2/output/x86_64-apple-darwin18/SHA256SUMS.part a828f7a623d38d549992af3035254ddde5a26e62638fe23a9bd76d2867683ce4 guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx-unsigned.dmg d24624964b6b68bfb03c88f49a22170f6fb0c16f4974949cde830be9b780b7f9 guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx-unsigned.tar.gz 7e4f0ca9b199a33acdabdfb7d6d6ae6524f1d5f3bb924951d55b792e3ad19dad guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx64.tar.gz bb466bd2a42c2914bf70b3b27335123eccad39b569d334c6f69ff08459c8ee6b guix-build-58857560fcd2/output/x86_64-linux-gnu/SHA256SUMS.part 430bc8eb7506c1041db17b2393927f82be44921c00f4a698477ab3a7cdd31313 guix-build-58857560fcd2/output/x86_64-linux-gnu/bitcoin-58857560fcd2-x86_64-linux-gnu-debug.tar.gz ab04cde29a2febcad6f50f288f6bbf026fc1a2c3309cd8f6276548d79c1d422e guix-build-58857560fcd2/output/x86_64-linux-gnu/bitcoin-58857560fcd2-x86_64-linux-gnu.tar.gz 014a940eadbcba286cac1f2df5e6d6ea1d21a3623d98d8b1300ea44ac0e9efc5 guix-build-58857560fcd2/output/x86_64-w64-mingw32/SHA256SUMS.part 99a38df35244d34ae8d247953e6f5903713d6098e66a5a41d509074dc216b926 guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win-unsigned.tar.gz a981dcde57666283d6b3bb9b0bbe2b1e54cb2b25b7301f3ecd3c5819acfd9823 guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64-debug.zip 862d38ba5b4120fb55f9b4a664379c1e2ccc2940bc54e46bb8884e9f77245e81 guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64-setup-unsigned.exe 539046ae22daf063c96fefe88359b9d3a85ea5f37fd27c56abd5bcb98490e290 guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64.zip

—

**Comment by hebasto** [comment]
#### Guix builds:

```
$ find guix-build-$(git rev-parse --short=12 HEAD)/output/ -type f -print0 | env LC_ALL=C sort -z | xargs -r0 sha256sum
670bc438033084003d015f23ed58cc92d065c0c962d9cc753104d32b250e0a1d  guix-build-58857560fcd2/output/aarch64-linux-gnu/SHA256SUMS.part
f27f69c7e77538aa01713f202d4db89f413c55204cd1fc99ed31f2081d564116  guix-build-58857560fcd2/output/aarch64-linux-gnu/bitcoin-58857560fcd2-aarch64-linux-gnu-debug.tar.gz
390c3166f4fa47b15a4dcbfa5205d69458fb8a1c1862de956f850d047782d816  guix-build-58857560fcd2/output/aarch64-linux-gnu/bitcoin-58857560fcd2-aarch64-linux-gnu.tar.gz
d6c12f3a1b22a4cea299f660c59a50c51061594d19ef0e087583dbece913cd69  guix-build-58857560fcd2/output/arm-linux-gnueabihf/SHA256SUMS.part
c04dc4071b2295f74b2381850a361550e481a99851bc435a184f52cd85cd5ed0  guix-build-58857560fcd2/output/arm-linux-gnueabihf/bitcoin-58857560fcd2-arm-linux-gnueabihf-debug.tar.gz
56ebdfb44035d6dd4d90d5445386deefd2d27d65d1bd2c75b69317f6e951b18a  guix-build-58857560fcd2/output/arm-linux-gnueabihf/bitcoin-58857560fcd2-arm-linux-gnueabihf.tar.gz
facf87168d70a7ace1d1b368ce2bb58e1390d1ea78e6850a2bef5e31d55b3c7d  guix-build-58857560fcd2/output/dist-archive/bitcoin-58857560fcd2.tar.gz
895f1d8006935500fd2fdcc9bb994270059b881cad37ebd0654bfa5800fed113  guix-build-58857560fcd2/output/powerpc64-linux-gnu/SHA256SUMS.part
b1a422795ac9f8f16b3e52c23861f617e49ef74fdc2c6ebaef6af906f04fff81  guix-build-58857560fcd2/output/powerpc64-linux-gnu/bitcoin-58857560fcd2-powerpc64-linux-gnu-debug.tar.gz
8b42e3a20dba091011bee1cb72e9f494f88ea781c94124e1f70fd2c22797cbb8  guix-build-58857560fcd2/output/powerpc64-linux-gnu/bitcoin-58857560fcd2-powerpc64-linux-gnu.tar.gz
6d67c99cc4beccc8ea48e579b8b4f75dc47a91597e335074d806535c8def958b  guix-build-58857560fcd2/output/powerpc64le-linux-gnu/SHA256SUMS.part
f366d52438137962322a4883c1883666e97169488f0b9ddc21ce067e3944396c  guix-build-58857560fcd2/output/powerpc64le-linux-gnu/bitcoin-58857560fcd2-powerpc64le-linux-gnu-debug.tar.gz
7cf04fa623e217bcccbb42220487f298009d460dd5cb471a8f3bae22ef3e9802  guix-build-58857560fcd2/output/powerpc64le-linux-gnu/bitcoin-58857560fcd2-powerpc64le-linux-gnu.tar.gz
3a9a8990aeb0c56a7b584d1823886c73990645770659a63c45e4b85d1c3f8d65  guix-build-58857560fcd2/output/riscv64-linux-gnu/SHA256SUMS.part
5f8f663779b3f7196da9f1c9b40be7b2a8e2b33fdb6baf673520c0b07b23ce26  guix-build-58857560fcd2/output/riscv64-linux-gnu/bitcoin-58857560fcd2-riscv64-linux-gnu-debug.tar.gz
3f43cf17a71e05506822fdbacfd8b7f3af9c739fd131842014dd80c92062d566  guix-build-58857560fcd2/output/riscv64-linux-gnu/bitcoin-58857560fcd2-riscv64-linux-gnu.tar.gz
3f89ecc812a1cd0d6009a119957b9bc5831f37249153b5fafe3268e4dd3bb59c  guix-build-58857560fcd2/output/x86_64-apple-darwin18/SHA256SUMS.part
a828f7a623d38d549992af3035254ddde5a26e62638fe23a9bd76d2867683ce4  guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx-unsigned.dmg
d24624964b6b68bfb03c88f49a22170f6fb0c16f4974949cde830be9b780b7f9  guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx-unsigned.tar.gz
7e4f0ca9b199a33acdabdfb7d6d6ae6524f1d5f3bb924951d55b792e3ad19dad  guix-build-58857560fcd2/output/x86_64-apple-darwin18/bitcoin-58857560fcd2-osx64.tar.gz
bb466bd2a42c2914bf70b3b27335123eccad39b569d334c6f69ff08459c8ee6b  guix-build-58857560fcd2/output/x86_64-linux-gnu/SHA256SUMS.part
430bc8eb7506c1041db17b2393927f82be44921c00f4a698477ab3a7cdd31313  guix-build-58857560fcd2/output/x86_64-linux-gnu/bitcoin-58857560fcd2-x86_64-linux-gnu-debug.tar.gz
ab04cde29a2febcad6f50f288f6bbf026fc1a2c3309cd8f6276548d79c1d422e  guix-build-58857560fcd2/output/x86_64-linux-gnu/bitcoin-58857560fcd2-x86_64-linux-gnu.tar.gz
014a940eadbcba286cac1f2df5e6d6ea1d21a3623d98d8b1300ea44ac0e9efc5  guix-build-58857560fcd2/output/x86_64-w64-mingw32/SHA256SUMS.part
99a38df35244d34ae8d247953e6f5903713d6098e66a5a41d509074dc216b926  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win-unsigned.tar.gz
a981dcde57666283d6b3bb9b0bbe2b1e54cb2b25b7301f3ecd3c5819acfd9823  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64-debug.zip
862d38ba5b4120fb55f9b4a664379c1e2ccc2940bc54e46bb8884e9f77245e81  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64-setup-unsigned.exe
539046ae22daf063c96fefe88359b9d3a85ea5f37fd27c56abd5bcb98490e290  guix-build-58857560fcd2/output/x86_64-w64-mingw32/bitcoin-58857560fcd2-win64.zip
```

**Comment by laanwj** [comment]
Code review ACK 58857560fcd299ae4aa03250298acfe63b26c305
Agree with @kristapsk  the include-change though—it can't be related to the preprocessor logic change.

File: src/bitcoin-cli.cpp (hunk 1 above)
Comment (fanquake) [reply src/bitcoin-cli.cpp]: Removed.

**Comment by fanquake** [comment]
> would make sense to split to a separate commit.

I've just dropped it for now.

**Review by kristapsk — APPROVED** [review]
ACK 6045a1464252075f4135bd4a69d202d55d124eb2
